src/NativeScript/ObjC/TNSArrayAdapter.mm

```diff
@@ -73,4 +74,13 @@ - (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState*)state objects
     return count;
 }
 
+- (void)dealloc {
+    {
+        JSLockHolder lock(self->_execState);
```

Comment:
`ObjCWrapperObject::~ObjCWrapperObject()` is a `JSCell` destructor and that is invoked during garbage collection where it is guaranteed that the API lock on the virtual machine is acquired.